ocaml/utils/zero_alloc_annotations.ml

```diff
@@ -0,0 +1,29 @@
+type t = Check_default | Check_all | Check_opt_only | No_check
```

Comment:
Maybe they're silly, but I like the file headers in the rest of the compiler. At a minimum, they're useful in identifying Jane Street as the source. Not sure what name should go in, though.

Reply:
https://github.com/ocaml-flambda/flambda-backend/blob/main/backend/cfg/cfg_stack_checks.mli is an example of a suitable header (JS-only files are supposed to be MIT-licensed, although this isn't consistently reflected across the tree yet).  You don't need to name a particular author, though it's probably @gretay-js